### PR TITLE
Change for anonymous sign-in to do not challenge

### PR DESCRIPTION
**Pull request by aji13187**
# Description

Issue: When we first try to `sign in`, then close it and then we try `anonymous` sign-in it will challenge you.

Fix: reset the portal so that the anonymous sign-in do not challenge in above scenario. 

## Type of change

- [x] Bug fix
- [ ] New sample implementation
- [ ] Sample viewer enhancement
- [ ] Other enhancement

**Platforms tested on**:

- [ ] Windows
- [ ] Android
- [ ] Linux
- [x] macOS
- [ ] iOS

## Checklist

- [x] Runs and compiles on all active platforms as a standalone sample
- [x] Runs and compiles in the sample viewer(s)
- [ ] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [x] Code is commented with correct formatting (CTRL+i)
- [x] All variable and method names are camel case
- [x] There is no leftover commented code
- [ ] Screenshots are correct size and display in description tab (500px by 500px, platform agnostic)
- [ ] If adding a new sample, it is added to the sample viewer
- [ ] Cherry-picked to Main branch (if applicable)

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/ShowOrgBasemaps/ShowOrgBasemaps.h [hunk 1]
```diff
@@ -64,10 +63,11 @@ class ShowOrgBasemaps : public QQuickItem
 
 private:
   void load();
+  void connectLoadStatusSignal();
 
   Esri::ArcGISRuntime::Map* m_map = nullptr;
   Esri::ArcGISRuntime::MapQuickView* m_mapView = nullptr;
-  Esri::ArcGISRuntime::Portal* m_portal = nullptr;
+  std::shared_ptr<Esri::ArcGISRuntime::Portal> m_portal = nullptr;
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/ShowOrgBasemaps/ShowOrgBasemaps.h:70]: This seems unrelated to solving the problem at hand. Typical usage is raw pointers and specifying `this` as the parent.

File: ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/ShowOrgBasemaps/ShowOrgBasemaps.cpp [hunk 2]
```diff
@@ -113,20 +110,21 @@ QString ShowOrgBasemaps::mapLoadError() const
...
 {
-  if (!m_portal)
-    return;
+  m_portal.reset(new Portal(this));
+  connectLoadStatusSignal();
 
-  if (anonymous)
-    load();
-  else {
+  if (!anonymous && m_portal) {
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/ShowOrgBasemaps/ShowOrgBasemaps.cpp:116]: Use consistent bracket placement style in .cpp files.

File: ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/ShowOrgBasemaps/ShowOrgBasemaps.h (hunk 1 above)
Comment (aji13187) [reply ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/ShowOrgBasemaps/ShowOrgBasemaps.h]: Used raw pointer.

File: ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/ShowOrgBasemaps/ShowOrgBasemaps.cpp (hunk 2 above)
Comment (aji13187) [reply ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/ShowOrgBasemaps/ShowOrgBasemaps.cpp]: Done.

**Review by JamesMBallard — APPROVED** [review]
